framework/src/actions/MeshOnlyAction.C

```diff
@@ -67,14 +69,92 @@ MeshOnlyAction::act()
+    auto & output_mesh = mesh_ptr->getMesh();
+    ExodusII_IO exio(output_mesh);
 
     Exodus::setOutputDimensionInExodusWriter(exio, *mesh_ptr);
 
     // Default to non-HDF5 output for wider compatibility
     exio.set_hdf5_writing(false);
 
     exio.write(mesh_file);
+
+    // Check if extra element integers should be outputted to Exodus file
+    unsigned int n_eeid = output_mesh.n_elem_integers();
+    if (n_eeid > 0)
+    {
+      // Iterate through all actions and see if `Outputs/output_extra_element_ids = true` in input
+      // file
+      const auto & output_actions = _app.actionWarehouse().getActionListByName("add_output");
+      bool output_extra_ids = true;
+      bool restrict_element_id_names = false;
+      std::vector<std::string> element_id_names;
+      for (const auto & act : output_actions)
+      {
+        // Extract the Output action
+        AddOutputAction * action = dynamic_cast<AddOutputAction *>(act);
+        if (!action)
+          continue;
+
+        InputParameters & params = action->getObjectParams();
+        if (params.isParamSetByUser("output_extra_element_ids"))
+        {
+          output_extra_ids = params.get<bool>("output_extra_element_ids");
+          if (output_extra_ids)
+          {
+            // `Outputs/extra_element_ids_to_output` sets a subset of extra element ids that should
+            // be outputted to Exodus
+            restrict_element_id_names = params.isParamValid("extra_element_ids_to_output");
+            if (restrict_element_id_names)
+              element_id_names =
+                  params.get<std::vector<std::string>>("extra_element_ids_to_output");
+            break;
+          }
+        }
+      }
+
+      if (output_extra_ids)
+      {
+        // Invoke ExodusII_IO_Helper to output extra element ids to Exodus file
+        auto & exio_helper = exio.get_exio_helper();
+
+        // Output empty timestep to Exodus file
+        int empty_timestep = 1;
+        Real default_time = 1.0;
+        exio_helper.write_timestep(empty_timestep, default_time);
+
+        // Retrieve extra element id names and associated data
+        std::vector<std::string> eeid_vars;
+        std::vector<Number> eeid_soln;
+        for (unsigned int i = 0; i < n_eeid; i++)
+        {
+          const auto eeid_name = output_mesh.get_elem_integer_name(i);
+          // If `Outputs/extra_element_ids_to_output` is not set, output all ids to Exodus
+          // Otherwise only output if the extra id name is contained within
+          // `Outputs/extra_element_ids_to_output`
+          if (!restrict_element_id_names ||
+              (std::find(element_id_names.begin(), element_id_names.end(), eeid_name) !=
+               element_id_names.end()))
+          {
+            eeid_vars.push_back(output_mesh.get_elem_integer_name(i));
+            for (auto & elem : output_mesh.element_ptr_range())
+              eeid_soln.push_back(elem->get_extra_integer(i));
```

Comment:
you need to resize and place.
push_back will trigger plenty of resizing

Reply:
Sorry I am unresolving this. I actually couldn't find a way to iterate through all elements in libmesh given the total number of elements in the mesh (`output_mesh.n_elem()`), so I had to resort to using a temporary variable (`soln_index`) in order to do this. If there is a better way to do this without the temporary variable, do let me know